gateway/src/shard/processor/session.rs

```diff
@@ -130,6 +127,24 @@ impl Session {
     pub fn set_heartbeat_interval(&self, new_heartbeat_interval: u64) {
         self.heartbeat_interval
             .store(new_heartbeat_interval, Ordering::Release);
+
+        #[allow(
+            clippy::cast_precision_loss,
+            clippy::cast_sign_loss,
+            clippy::cast_possible_truncation
+        )]
+        let heartbeats_per_120s = (120_000.0 / new_heartbeat_interval as f64).ceil() as usize;
```

Comment:
Is there a way we can avoid any of these allows?

Reply:
Done